### ISSUE TITLE
[ANDROID] Methods addSQLiteSuffix() and deleteOldDatabases() fails if 'databases' folder are missing

### ISSUE THREAD

**Issue by dragermrb** [capacitor-community/sqlite#147]
**Describe the bug**
On Android, `addSQLiteSuffix()` and `deleteOldDatabases()` methods don't work if **'databases'** folder are missing.

We are migrating an application from Cordova to Capacitor3, so we tried to do DB migration right after initializing the plugin. When installing the app on a new device, the **'databases'** folder does not exist and the indicated methods fail.

We have not tested with IOS yet

**To Reproduce**
Try to add suffix on empty device before create any DB.

```typescript
export class AppComponent {
  constructor(
    private platform: Platform,
    private sqliteService: SQLiteService
  ) {
    this.platform.ready().then(() => {
      this.sqliteService.initializePlugin();
    });
  }
}
```

```typescript
export class SQLiteService {
  platform: string;
  sqlite: SQLiteConnection;

  constructor() {}

  /**
   * Plugin Initialization
   */
  initializePlugin(): Promise<boolean> {
    return new Promise(async (resolve) => {
      const sqlitePlugin: any = CapacitorSQLite;
      this.sqlite = new SQLiteConnection(sqlitePlugin);

      try {
        /*
            this.addSQLiteSuffix() and this.deleteOldDatabases() are the same as 
                on https://github.com/jepiqueau/angular-sqlite-app-starter
        */

        this.addSQLiteSuffix().then(
            () => this.deleteOldDatabases(),
            (error) => console.error('addSQLiteSuffix', error)
        );
      } catch (error) {
        console.error('catch', error);
      }

      resolve(true);
    });

    // ...
  }
```

On DevTools console (we can't catch the `Unhandled Promise rejection`):

```
Unhandled Promise rejection: addSQLiteSuffix: Folder /data/user/0/com.testapp.id/databases does not exist ; Zone: <root> ; Task: null ; Value: Error: addSQLiteSuffix: Folder /data/user/0/com.testapp.id/databases does not exist
    at Object.cap.fromNative ((index):429)
    at <anonymous>:1:18 Error: addSQLiteSuffix: Folder /data/user/0/com.testapp.id/databases does not exist
    at Object.cap.fromNative (http://192.168.0.200:8100/:429:32)
    at <anonymous>:1:18
```

**Expected behavior**
That there are no errors if there are no DBs to migrate or to be able to capture them

**Smartphone (please complete the following information):**

- Device: Pixel 4 Emulator
- OS: Android API 29

**Comment by jepiqueau** [comment]
@dragermrb thanks for using the plug-in. Where are your Cordova sqlite databases stored that you want to migrate?.

**Comment by dragermrb** [comment]
There are two different cases:

- Users with the previous app (cordova) who update to the new app (Capacitor3). These users have the databases in the default folder, **everything works correctly**.
- Users installing the app for the first time (the new app with Capacitor). When executing the `addSQLiteSuffix()` method error occurs because the folder **'databases'** does not exist (it is a new installation) and there is nothing to migrate.

One option is to run `addSQLiteSuffix()` only when there are DBs to migrate (how can we check if they exist?). The other option is for `addSQLiteSuffix()` to silently (without promise reject) terminate if there are no DBs to migrate.

**Comment by jepiqueau** [comment]
@dragermrb i never imagine the case 2 as the command has been made for peopel migrating for cordova. anyhow i will provide a fix for this
on Android
```java
    public void addSQLiteSuffix(Context context, String folderPath) throws Exception {
        String pathDB = new File(context.getFilesDir().getParentFile(), "databases").getAbsolutePath();
        File dirDB = new File(pathDB);
        if (!dirDB.isDirectory()) {
            dirDB.mkdir();
        }
        String pathFiles = this.getFolder(context, folderPath);
        // check if the path exists
        File dir = new File(pathFiles);
        if (!dir.exists()) {
            throw new Exception("Folder " + dir + " does not exist");
        }
        String[] listFiles = dir.list();
        if(!pathDB.equals(pathFiles) && listFiles.length == 0) {
            throw new Exception("Folder " + dir + " no database files");
        }
        for (String file : listFiles) {
            if (uFile.getFileExtension((file)).equals("db")) {
                if (!file.contains("SQLite.db")) {
                    String fromFile = file;
                    String toFile = file.replace(".db", "SQLite.db");
                    boolean ret = uFile.copyFromNames(context, pathFiles, fromFile, pathDB, toFile);
                    if (!ret) {
                        String msg = "Failed in copy " + fromFile + " to " + file;
                        throw new Exception(msg);
                    }
                }
            }
        }
        return;
    }
```
i have still to look for iOS

**Comment by jepiqueau** [comment]
@dragermrb Fix in 3.1.2-1

**Comment by dragermrb** [comment]
Thanks @jepiqueau for the fast answer!
